pre-commit/package_app_dependencies.py

```diff
@@ -0,0 +1,155 @@
+"""
+Builds wheel files for the dependencies of an app, specified in requirements.txt, into the wheels/
+folder of the app repo, and updates the app's JSON config specifying any generated wheels as pip
+dependencies.
+
+NOTE: This script is meant to be executed from a manylinux2014_x86_64 container to ensure that any
+wheels built with pre-compiled binaries will be compatible on a CentOS 7/RHEL 7 system running Phantom.
+
+https://github.com/pypa/manylinux
+"""
+import json
+import logging
+import os
+import random
+import re
+import shutil
+import string
+import subprocess
+import sys
+from collections import namedtuple
+
+PIP_3_6_PATH = '/opt/python/cp36-cp36m/bin/pip'
+PLATFORM = 'manylinux2014_x86_64'
+REPAIRED_WHEELS_REL_PATH = 'repaired-wheels'
+
+WHEEL_PATTERN = re.compile(
+    '^(?P<distribution>([A-Z0-9][A-Z0-9._-]*[A-Z0-9]))-([0-9]+\\.[0-9]+\\.[0-9]+|[0-9]+\\.[0-9]+)-.+\\.whl$',
+    re.IGNORECASE)
+
+AppJson = namedtuple('AppJson', ['file_name', 'indent', 'content'])
+
+
+def load_app_json(app_dir):
+    json_files = [f for f in os.listdir(app_dir)
+                  if not f.endswith('.postman_collection.json') and f.endswith('.json')]
+
+    if len(json_files) != 1:
+        error_msg = 'Expected a single json file in {} but got {}'.format(app_dir, json_files)
+        logging.error(error_msg)
+        raise ValueError(error_msg)
+
+    with open(os.path.join(app_dir, json_files[0])) as f:
+        json_str = f.read()
+
+    start = idx = json_str.find('\n') + 2
+    while json_str[idx] != '"':
+        idx += 1
+    indent = idx - start + 1
```

Comment:
is this just to preserve indentation? not sure if it really matters since whitespace doesn't matter in JSON

also, iinm I think this will give the wrong result in the case of consecutive newlines?

Reply:
yep, I've simplified the code to just use a fixed indent (4 white spaces)